src/TaskRouter.cpp

```diff
@@ -31,28 +35,69 @@ bool RouterTask::setup(System &system) {
 bool RouterTask::loop(System &system) {
   // do routing
   if (!_fromModem.empty()) {
-    std::shared_ptr<APRSMessage> msg  = _fromModem.getElement();
-    String                       path = msg->getPath();
+    std::shared_ptr<APRSMessage> modemMsg  = _fromModem.getElement();
+    String path;
 
-    if (!(path.indexOf("RFONLY") != -1 || path.indexOf("NOGATE") != -1 || path.indexOf("TCPIP") != -1)) {
-      if (!path.isEmpty()) {
-        path += ",";
+    if (system.getUserConfig()->aprs_is.active && modemMsg->getSource() != system.getUserConfig()->callsign) {
+      std::shared_ptr<APRSMessage> aprsIsMsg = std::make_shared<APRSMessage>(*modemMsg);
```

Comment:
this is not a valid/good code. use the `modemMsg` directly or create a new copy with the `new` keyword.

Reply:
make_shared is perfect, sorry for the confusion!